tests/test_afwTable.py

```diff
@@ -0,0 +1,155 @@
+
+    def test_create_false(self):
+        """Test that NoRepositoryAtRoot is raised if no repository exists and create is False"""
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        with self.assertRaises(dafPersist.NoRepositroyAtRoot):
+            SqlStorage(dbLocation, create=False)
+
+    def test_write_read(self):
+        """Test that Butler can write a BaseCatalog to an sqlite database, and that the database can be read
+        by sqlalchemy and compares equal to the original.
+        """
+        cat_expected = _make_afw_base_catalog()
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        butler = dafPersist.Butler(outputs={'cfgRoot': self.testDir, 'root': dbLocation, 'mapper': MyMapper})
+
+        # Test writing first
+        butler.put(cat_expected, 'table')
+        del butler
+
+        # Test reading back with raw object
+        engine = sqlalchemy.create_engine(dbLocation)
+        rows = engine.execute("select a, b from testname")
+        # FIXME: Hopefully tables implement __eq__ in the future
+        self._compare_table(cat_expected, rows)
+
+        # Test reading back via butler.get
+        butler = dafPersist.Butler(inputs=self.testDir)
+        cat_reloaded = butler.get('table')
+        self._compare_table(cat_expected, cat_reloaded)
+
+    def test_no_cfg_root_raises_with_output_repo(self):
+        """Right now we don't support writing a RepositoryCfg direcetly into a database. Test that an
+        exception is raised if a cfgRoot is not specified for the database repository."""
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        with self.assertRaises(RuntimeError):
+            butler = dafPersist.Butler(outputs={'root': dbLocation, 'mapper': MyMapper})
+            del butler
+
+    def test_no_cfg_root_raises_with_input_repo(self):
+        """Right now we don't support reading a RepositoryCfg direcetly into a database. Test that an
+        exception is raised if a cfgRoot is not specified for the database repository."""
+        # create a database repo with a cfg on the filesystem:
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        butler = dafPersist.Butler(outputs={'cfgRoot': self.testDir, 'root': dbLocation, 'mapper': MyMapper})
+        del butler
+        # pass the dbLocation for root (instead of the cfgRoot) and check that it raises an exception.
+        # with self.assertRaises(RuntimeError):
+        with self.assertRaises(RuntimeError):
+            butler = dafPersist.Butler(inputs=dbLocation)
+            del butler
+
+    def _compare_table(self, afw_cat, rows):
+        columns = [str(i.field.getName()) for i in afw_cat.schema]
+        for cat1_row, cat2_row in zip(afw_cat, rows):
+            for column in columns:
+                self.assertEqual(cat1_row[column], cat2_row[column])
+
```

Comment:
_compare_table() is fine for what it does here.  Given it's testing, one simpler alternative might be to generate then compare the hashes(MD5?) of the 2 table contents(rows).